### DISCUSSION TITLE
Route revalidation

### DISCUSSION THREAD

**Discussion by yunior0814** [remix-run/react-router#9387] [Q&A]
How I can get a loading state in this scenario when the route is revalidate?

function Posts(){
  const data = useLoaderData();
  return  <Suspense fallback={<p>Loading layout</p>}>
          <Await resolve={data.response}>
            <Result />
          </Await>
        </Suspense>
}

<Route path="Post" loader={referLoaderData} element={<Posts />}  shouldRevalidate={({ currentUrl }) => {
              return currentUrl.pathname === "/comments/new";
            }} />
            
  <Route path="/comments/new"  action={newCommentAction} element={<Form action="/comments/edit" method="post">...</Form>} />

**Comment by tkapitein** [comment]
Not sure if this is what you are looking for, but I use the [useNavigation](https://reactrouter.com/en/main/hooks/use-navigation) hook.

**Reply by yunior0814** [reply]
no it doesn't work either

**Reply by yunior0814** [reply]
i found it, with useFetchers thanks anyway
